include/rcutils/error_handling.h

```diff
@@ -103,6 +107,25 @@ rcutils_set_error_state(
 #define RCUTILS_SET_ERROR_MSG(msg, allocator) \
   rcutils_set_error_state(msg, __FILE__, __LINE__, allocator);
 
+/// Set the error message using a format stirng and format arguments.
+/**
+ * This function sets the error message using the given format string and
+ * then frees the memory allocated during the string formatting.
+ *
+ * \param[in] allocator The allocator to be used when allocating space for the error state.
+ * \param[in] format_string The string to be used as the format of the error message.
+ * \param[in] ... Arguments for the format string.
+ */
+#define RCUTILS_SET_ERROR_MSG_WITH_FORMAT_STRING(allocator, format_string, ...) \
+  char * output_msg = NULL; \
+  output_msg = rcutils_format_string(allocator, format_string, __VA_ARGS__); \
+  if (output_msg) { \
+    RCUTILS_SET_ERROR_MSG(output_msg, allocator); \
+    allocator.deallocate(output_msg, allocator.state); \
+  } else { \
+    RCUTILS_SAFE_FWRITE_TO_STDERR("Failed to allocate memory for error message"); \
+  }
```

Comment:
This should probably be put into a scope, i.e. within something like `{}` or `do {} while (0)`, so that you can use this macro more than once in a function without colliding symbol definitions, e.g. `char * output_msg`.

Reply:
Sorry I didn't see this before.

Reply:
@wjwwood exactly, good catch :+1: